### PR TITLE
CRM-21574 NFC code cleanup of CRM_Friend_BAO_Friend class

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Towards #11425. This PR contains the non-functional code changes that cleanup that class.

Before
----------------------------------------
Less easy to maintain code

After
----------------------------------------
More easy to maintain code

Comments
----------------------------------------
Seems to be impossible to get a reviewer for #11425 as I think virtually no-one uses the PCP classes so let's try breaking it down a little.

---

 * [CRM-21574: Allow to disable sending of email from source contact for tell a friend on pcp](https://issues.civicrm.org/jira/browse/CRM-21574)

### CODE REVIEW

**Comment by mattwire** [comment]
Anyone other than @eileenmcnaughton able to review this PR please - it's just code cleanup... @seamuslee001 @michaelmcandrew @monishdeb ??

**Comment by michaelmcandrew** [comment]
not me at the moment, I'm afraid @mattwire :(

File: CRM/Friend/BAO/Friend.php [hunk 1]
```diff
@@ -111,14 +116,14 @@ public static function create(&$params) {
       }
     }
 
-    $frndParams = array();
-    $frndParams['entity_id'] = $params['entity_id'];
-    $frndParams['entity_table'] = $params['entity_table'];
-    self::getValues($frndParams);
+    $friendParams = array();
+    $friendParams['entity_id'] = $params['entity_id'];
+    $friendParams['entity_table'] = $params['entity_table'];
```
Comment (monishdeb) [inline CRM/Friend/BAO/Friend.php]: Change to 
```
$friendParams = [
 'entity_id' => $params['entity_id'],
 'entity_table' => $params['entity_table'],
];

File: CRM/Friend/Form.php [hunk 2]
```diff
@@ -45,14 +45,21 @@ class CRM_Friend_Form extends CRM_Core_Form {
...
    */
   protected $_entityId;
 
   /**
-   * The table name of the entity that we are proceessing.
+   * Tell a friend id in db.
+   *
+   * @var int
+   */
+  public $_friendId;
```
Comment (monishdeb) [inline CRM/Friend/Form.php]: will it be protected instead of public?

File: CRM/Friend/BAO/Friend.php [hunk 3]
```diff
@@ -167,28 +171,31 @@ public static function create(&$params) {
...
 
-    //process sending of mails
+    // Process sending of mails
     $mailParams['title'] = CRM_Utils_Array::value('title', $params);
-    $mailParams['general_link'] = CRM_Utils_Array::value('general_link', $frndParams);
+    $mailParams['general_link'] = CRM_Utils_Array::value('general_link', $friendParams);
     $mailParams['message'] = CRM_Utils_Array::value('suggested_message', $params);
 
     // get domain
     $domainDetails = CRM_Core_BAO_Domain::getNameAndEmail();
```
Comment (monishdeb) [inline CRM/Friend/BAO/Friend.php]: Can you change it to 
```patch
-  $domainDetails = CRM_Core_BAO_Domain::getNameAndEmail();
+ list($mailParams['email_from'], $_) = CRM_Core_BAO_Domain::getNameAndEmail();
...
- // Default "from email address" is default domain address.
- // This is normally overridden by one of the if statements below
- $mailParams['email_from'] = $domainDetails[1];

File: CRM/Friend/Form.php (hunk 2 above)
Comment (mattwire) [reply CRM/Friend/Form.php]: It's used in public static function buildFriendForm($form) which is called from two other classes (Contribute and Event) so it needs to be public unless we do a lot more work, at which point we would risk breaking things :-)

File: CRM/Friend/BAO/Friend.php (hunk 3 above)
Comment (mattwire) [reply CRM/Friend/BAO/Friend.php]: Thanks - nice cleanup.

File: CRM/Friend/BAO/Friend.php (hunk 1 above)
Comment (mattwire) [reply CRM/Friend/BAO/Friend.php]: Yes

**Comment by mattwire** [comment]
Thanks for reviewing @monishdeb I've made changes as requested.

**Comment by monishdeb** [comment]
(*CiviCRM Review Template WORD-1.1*)



* ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) __PASS__
* ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __PASS__
* ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __PASS__
* ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __PASS__
* ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __PASS__
* ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) __PASS__ Tested on local, works fine
* ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __PASS__
* ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __PASS__

**Pull request merged**